src/bin/cog_commit.rs

```diff
@@ -1,18 +1,76 @@
 use std::fmt::Write;
 
-use cocogitto::COMMITS_METADATA;
-
 use anyhow::{bail, Result};
-use conventional_commit_parser::commit::Separator;
+use colored::Colorize;
+use conventional_commit_parser::commit::{CommitType, Separator};
 use itertools::Itertools;
 
+use cocogitto::settings::Settings;
+use cocogitto::{COMMITS_METADATA, SETTINGS};
+
 pub fn commit_types() -> Vec<&'static str> {
     COMMITS_METADATA
         .iter()
         .map(|(commit_type, _)| commit_type.as_ref())
         .collect()
 }
 
+pub fn explain_commit_type(commit_type: &str) -> String {
+    let semantic_versioning_minor_msg = format!(
+        "{} : this correlates with {}",
+        "Semantic versioning".italic(),
+        "MINOR".bold()
+    );
+    let semantic_versioning_patch_msg = format!(
+        "{} : this correlates with {}",
+        "Semantic versioning".italic(),
+        "PATCH".bold()
+    );
+    let semantic_versioning_no_correlation_msg =
+        format!("{} : no correlation", "Semantic versioning".italic());
+
+    let commit_type = CommitType::from(commit_type);
+
+    match commit_type {
+        CommitType::BugFix => {
+            get_formatted_explain_msg(&commit_type, &semantic_versioning_patch_msg)
+        }
+        CommitType::Feature => {
+            get_formatted_explain_msg(&commit_type, &semantic_versioning_minor_msg)
+        }
+        _ => get_formatted_explain_msg(&commit_type, &semantic_versioning_no_correlation_msg),
+    }
+}
+
+fn get_formatted_explain_msg(
+    commit_type: &CommitType,
+    semantic_versioning_correlation_msg: &str,
+) -> String {
+    let learn_more_msg = "Learn more about supported commit types :\r
```

Comment:
This whole text should probably be kept outside of the function as a `const` instead, also I don't think the manual `\r`s are necessary